### ISSUE TITLE
__init__ method is not called on creation

### ISSUE THREAD

**Issue by geiseri** [jsonpickle/jsonpickle#146]
On restoring from JSON the objects are created but properties that are defined in __init__ are not created if they are missing in the JSON.  I am not sure if this is intended though.  Is there a way to define the behavior of missing properties in the object?

**Comment by davvid** [comment]
If you have a unit case we can add to the test suite it might help illuminate the issue.

Generally, if you need to special-case the handling of a specific class then it can either implement the `__getstate__(self) -> data` and `__setstate__(self, data)` protocol, or you can register a custom handler.

**Comment by GuillemGSubies** [comment]
I have the same problem. I think I found what happens. If I specify **every** parameter  that the init method has, everything works correctly, but if I don't, jsonpickle instances them to `None`

These are the defaul scikit-learn parameters:

```python
jsonpickle.decode('''{
    "py/object": "sklearn.ensemble.AdaBoostRegressor",
    "py/state": {
        "base_estimator": null,
        "n_estimators": 50,
        "learning_rate": 1.0,
        "loss": "linear",
        "random_state": null
}}''')
```

>>> AdaBoostRegressor(base_estimator=None, learning_rate=1.0, loss='linear',
                  n_estimators=50, random_state=None)

However if I do this:

```python
jsonpickle.decode('''{"py/object": "sklearn.ensemble.AdaBoostRegressor"}''')
```

I get all Nones:

>>> jsonpickle.decode('''{"py/object": "sklearn.ensemble.AdaBoostRegressor"}''')

AdaBoostRegressor(base_estimator=None, learning_rate=None, loss=None,
                  n_estimators=None, random_state=None)

I understand the behaviour of this, jsonpickle does not initialize the object, it unpickles it as if it were already initialized and saved.

However I think my problem is a kinda common use case and it would be really cool if we had the chance to tell jsonpickle.decode that we want it to call the init method instead of trying to load an object.
